src/Common/src/System/Net/CookieParser.cs

```diff
@@ -542,6 +543,18 @@ internal string GetString()
             return _tokenizer.GetCookieString();
         }
 
+        private static void SetCookieName(Cookie cookie, string name)
+        {
+            try
+            {
+                cookie.Name = name;
+            }
+            catch (CookieException)
+            {
+                Debug.Assert(cookie.Name == string.Empty);
+            }
```

Comment:
And if string.Empty will always be rejected, should we special-case it to avoid the exception being thrown?